universal-application-tool-0.0.1/app/controllers/admin/AdminApplicationController.java

```diff
@@ -57,8 +60,29 @@ public AdminApplicationController(
     this.profileUtils = checkNotNull(profileUtils);
     this.applicationView = checkNotNull(applicationView);
     this.applicationRepository = checkNotNull(applicationRepository);
-    this.clock = clock;
+    this.clock = checkNotNull(clock);
     this.exporterService = checkNotNull(exporterService);
+    this.jsonExporter = checkNotNull(jsonExporter);
+  }
+
+  /** Download a JSON file containing all applications to all versions of the specified program. */
+  @Secure(authorizers = Authorizers.Labels.ANY_ADMIN)
+  public Result downloadAllJson(Http.Request request, long programId) {
```

Comment:
maybe not for this PR but would love to get a lil load testing on these

Reply:
yeah it'd be good to have some load testing of any kind, really